epochStart/bootstrap/process.go

```diff
@@ -519,10 +519,14 @@ func (e *epochStartBootstrap) computeIfCurrentEpochIsSaved() bool {
 
 	roundsSinceEpochStart := computedRound - int64(e.baseData.epochStartRound)
 	log.Debug("epoch start round", "round", e.baseData.epochStartRound, "roundsSinceEpochStart", roundsSinceEpochStart)
-	epochEndPlusGracePeriod := float64(e.generalConfig.EpochStartConfig.RoundsPerEpoch) * (gracePeriodInPercentage + 1.0)
+	epochEndPlusGracePeriod := float64(e.getRoundsPerEpoch()) * (gracePeriodInPercentage + 1.0)
 	return float64(roundsSinceEpochStart) < epochEndPlusGracePeriod
 }
 
+func (e *epochStartBootstrap) getRoundsPerEpoch() int64 {
+	return e.coreComponentsHolder.ChainParametersHandler().CurrentChainParameters().RoundsPerEpoch
```

Comment:
we might need to use chain paramaters for epoch specifically

Reply:
refactored